Xamarin.Forms.Xaml/DesignerInterface.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms.Xaml
+{
+	// IMPORTANT!!
+	//  READ BEFORE MODIFYING!!
+	// This class represents the API that the Forms previewer will use
+	//  via reflection. Adding members or changing implementations is fine, but
+	//  DO NOT break this API unless you know what you're doing and/or are
+	//  coordinating with the designer team.
+	// Thanks!
+	static class DesignerInterface
+	{
+		const int LatestSupportedVersion = 1;
+		static int Version => LatestSupportedVersion;
+
+		// These types are used via reflection for type checks..
+		static Type IMarkupExtensionType => typeof (IMarkupExtension);
+		static Type ViewType => typeof (Xamarin.Forms.View);
+		static Type BoxViewType => typeof (Xamarin.Forms.BoxView);
+		static Type VisualElementType => typeof (Xamarin.Forms.VisualElement);
+		static Type ApplicationType => typeof (Xamarin.Forms.Application);
+
+		static Func<ResourceLoader.ResourceLoadingQuery, ResourceLoader.ResourceLoadingResponse> ResourceProvider2 {
+			get => ResourceLoader.ResourceProvider2;
+			set => ResourceLoader.ResourceProvider2 = value;
+		}
+
+		static Action<Exception> ExceptionHandler {
+			get => ResourceLoader.ExceptionHandler;
+			set => ResourceLoader.ExceptionHandler = value;
+		}
+
+		static Func<IList<XamlLoader.FallbackTypeInfo>, Type, Type> FallbackTypeResolver {
+			get => XamlLoader.FallbackTypeResolver;
+			set => XamlLoader.FallbackTypeResolver = value;
+		}
+
+		static Action<XamlLoader.CallbackTypeInfo, object> ValueCreatedCallback  {
+			get => XamlLoader.ValueCreatedCallback;
+			set => XamlLoader.ValueCreatedCallback = value;
+		}
```

Comment:
Can we also make `FallbackTypeInfo` and `CallbackTypeInfo` classes instead of structs?